### PR TITLE
FIX #12216 issue add unit test

### LINKED ISSUE #12216: Add test case for SingleTableRuleBuilder

### PR DESCRIPTION
**Pull request by hoorf**
Fixes #12216

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SingleTableRuleBuilderTest {
+    static {
```
Comment (taojintianxia) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilderTest.java:43]: leave an empty line between static and class definination will be better . like : 

```java
public class SingleTableRuleBuilderTest {

    static {.....
```

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        ShardingSphereServiceLoader.register(SchemaRuleBuilder.class);
+    }
+
+    @Test
+    public void assertBuild() {
+        Collection<SchemaRuleBuilder> registeredServices = OrderedSPIRegistry.getRegisteredServices(SchemaRuleBuilder.class);
```
Comment (taojintianxia) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilderTest.java:49]: variable name is registeredServices, but it's a builder collection, why not registeredServiceBuilders ?

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SingleTableRuleBuilderTest {
```
Comment (tristaZero) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/rule/builder/SingleTableRuleBuilderTest.java:43]: Hi, @hoorf Thanks for your work, sorry to feedback a little late. But do you think it is better to define it as a final class as it is just for a test?

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Review by tristaZero — APPROVED** [review]
@hoorf  @taojintianxia 
hey, nice job, guys. ;-)

**Pull request merged**